### ISSUE TITLE
delimiter-dangle triggers on interface

### ISSUE THREAD

**Issue by SimeonC** [gajus/eslint-plugin-flowtype#381]
In an `interface` I don't think this should trigger the `delimiter-dangle` rule. Mostly because `prettier` keeps adding in the semi-colon and in this situation it is more like a class definition than an object definition so I think the semi-colon on `renderNode` should be there.

```js
interface RenderMarkInterface {
  renderMark: (props: *) => Element<*> | Node | null;
  renderNode?: null; // This line throws "Unexpected trailing delimiter (flowtype/delimiter-dangle)
}
```

**Comment by gajus** [comment]
PR is welcome.

**Comment by SimeonC** [comment]
Took me ages but I finally got around to creating a PR for this. Ended up adding a second argument to `delimiter-dangle` so the behaviour of `interface` types can be properly controlled. Turns out that the AST parser thinks that an `interface` is an `Object` annotation, whereas it's closer to a `Class` annotation.
